### PR TITLE
Fix empty issue dropdown in Attach to Jira Issue modal (MM-67234)

### PR DESCRIPTION
**Pull request by nang2049**
### Summary
- Fix the "Attach to Jira Issue" modal dropdown not showing any issues when first opened
- When the query string is empty use `ORDER BY updated DESC` to return recently updated issues instead of constructing invalid JQL
- Add null guard for the search results data on the frontend to prevent crashes when the API returns no data

### Ticket
[MM-67234](https://mattermost.atlassian.net/browse/MM-67234)

### QA Steps
1. Connect a Mattermost account to a Jira instance
2. Post a message in any channel
3. Click the post menu then "Attach to Jira Issue"
4. Select your Jira instance
5. **Before fix**: The issue dropdown shows "No options" when first opened
6. **After fix**: The issue dropdown shows recently updated issues immediately when opened

[MM-67234]: https://mattermost.atlassian.net/browse/MM-67234?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

## Change Impact: 🟡 Medium

**Reasoning:** The changes touch both backend and frontend layers (JQL construction, Jira client error handling, and UI null-guards) and add a Jira Cloud-specific client implementation; they are focused and targeted but affect request/response flows across integration boundaries.

**Regression Risk:** Moderate. The fixes reduce errors for empty queries and prevent frontend crashes, but the new cloud-specific SearchIssues implementation and altered error-path handling broaden the surface area for integration differences and could cause divergent behavior across Jira deployments (reviewer evidence shows Cloud-specific verification still required).

**QA Recommendation:** Full QA across environments is recommended — verify behavior on both Jira Server and Jira Cloud, confirm the dropdown shows recently updated issues on open and while typing, and test error/timeout/unauthorized responses to ensure graceful UI handling. Prioritize manual verification on Cloud and add automated tests for JQL fallback, escaping, and nil-response handling; do not skip manual QA.
*Generated by CodeRabbitAI*

### CODE REVIEW

**Review by avasconcelos114 — APPROVED** [review]
LGTM

**Comment by nang2049** [comment]
@coderabbitai review

File: webapp/src/components/jira_issue_selector/jira_issue_selector.jsx [hunk 1]
```diff
@@ -60,12 +60,16 @@ export default class JiraIssueSelector extends Component {
         };
 
         return this.props.searchIssues(params).then(({data}) => {
+            if (!data) {
+                return [];
+            }
             return data.map((issue) => ({
                 value: issue.key,
                 label: `${issue.key}: ${issue.fields.summary}`,
             }));
```
[the remark this answers is not in the document]
Comment (nang2049) [reply webapp/src/components/jira_issue_selector/jira_issue_selector.jsx:69]: This is a false positive. `doFetchWithResponse` in `webapp/src/client/index.ts` wraps the parsed JSON response into `{response, data}` before returning so `({data})` correctly destructures the array.

**Comment by nang2049** [comment]
@coderabbitai review

**Comment by nang2049** [comment]
@coderabbitai approve

**Comment by ogi-m** [comment]
Testing with Jira Cloud, I still don't get any issues in the dropdown until I enter an issue key:

https://github.com/user-attachments/assets/ff709262-f10a-40c5-8257-311d39a3ece2

**Comment by nang2049** [comment]
@ogi-m fixed

**Comment by nang2049** [comment]
@coderabbitai review

**Comment by nang2049** [comment]
@coderabbitai review

**Review by ogi-m — APPROVED** [review]
LGTM, the most recently updated issues are now shown

**Comment by nang2049** [comment]
@coderabbitai approve
